theia-rust-docker/Dockerfile

```diff
@@ -0,0 +1,54 @@
+#Can't use node image, because it creates a user that conflicts
+FROM ubuntu:18.04
+
+RUN apt-get update -yq \
+    && apt-get install curl gnupg build-essential gcc g++ gdb make python sudo git -yq 
+
+#allow a user to become root in the container
+ADD sudoers /etc/sudoers
+
+ARG GITHUB_TOKEN
+ARG NODE_OPTIONS="--max-old-space-size=4096"
+ARG NODE_VERSION=10
+
+ADD install_node.sh /root/install_node.sh
+RUN bash /root/install_node.sh --version $NODE_VERSION
+ADD nvm_setup.sh /root/nvm_setup.sh
+
+#install theia
+RUN mkdir /root/theia_rust_extension
+RUN . /root/nvm_setup.sh && cd /root/theia_rust_extension && yarn global add node-gyp
+ADD next.package.json /root/theia_rust_extension/package.json
+RUN git clone https://github.com/eclipse/theia-cpp-extension.git /root/theia-cpp-extension
+RUN cp -R /root/theia-cpp-extension/packages /root/theia_rust_extension/.
+RUN . /root/nvm_setup.sh && cd /root/theia_rust_extension && yarn --cache-folder /root/theia_rust_extension/ycache 
+RUN . /root/nvm_setup.sh && cd /root/theia_rust_extension/packages/cpp-debug && yarn --cache-folder /root/theia_rust_extension/ycache 
+RUN . /root/nvm_setup.sh && cd /root/theia_rust_extension && yarn theia build
+
+# when creating a container a volume is mapped to /root/rust
+# to allow to make persistent changes
+ARG rust_channel=nightly
+ENV RUSTUP_HOME /root/rust/rustup
+ENV CARGO_HOME /root/rust/cargo
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > /tmp/sh.rustup.rs && sh /tmp/sh.rustup.rs -y
+
+#install language server dependencies for theia
+RUN . /root/rust/cargo/env && rustup toolchain install $rust_channel
+RUN . /root/rust/cargo/env && rustup default $rust_channel
+RUN . /root/rust/cargo/env && rustup component add rust-src rust-docs rls rust-analysis rustfmt clippy
+
+
+RUN . /root/rust/cargo/env && cargo +$rust_channel install racer
+
+EXPOSE 3000
+ENV SHELL /bin/bash
+# script to start theia
+ADD run.sh /root/run.sh
+# script to set environment
+ADD bashrc /root/bashrc
+RUN chmod 755 /root \
+    && chmod 755 /root/bashrc
+WORKDIR /root/theia_rust_extension
+CMD [ "yarn", "theia", "start", "/home/project", "--hostname=0.0.0.0" ]
```

Comment:
I was able to make it work by using this entrypoint, specifying the absolute path for the node executable:
`ENTRYPOINT [ "/root/.nvm/versions/node/v10.16.0/bin/node", "/root/theia_rust_extension/src-gen/backend/main.js", "/home/project", "--hostname=0.0.0.0" ]`

It's a bit fragile however, since the node path will change with the node version. 

Without the full path, node is not found and so Theia can't start. I  think maybe the `nvm` configuration is not active when the entrypoint is executed. If I instead override the entrypoint with `bash`, then node is in the path, presumably because root's `.bashrc` is read and configures `.nvm`.

Reply:
I fixed the issue by calling nvm_setup as part of /root/bashrc. The tests succeed now. 

I also added a commit_screenshots.sh script. This scipt is called when a test failure occurs. It will commit all screenshot pngs to a 'bugs' branch. This branch is emptied at every push.  If you want to use this script a separate empty branch has to be created to which travis can commit the screenshots.

Please let me know if you need any other rework.

Reply:
> I fixed the issue by calling nvm_setup as part of /root/bashrc. The tests succeed now.

Confirmed: now works locally for me